src/api/cow/cow.spec.ts

```diff
@@ -0,0 +1,441 @@
+    {
+      body: undefined,
+      headers: {
+        'Content-Type': 'application/json',
+        'X-AppId': '0x0000000000000000000000000000000000000000000000000000000000000000',
+      },
+      method: 'GET',
+    }
+  )
+})
+
+test('Invalid: Get Profile Data from unexisting address', async () => {
+  fetchMock.mockResponseOnce(
+    JSON.stringify({
+      errorType: 'NotFound',
+      description: "You've passed an invalid URL",
+    }),
+    { status: HTTP_STATUS_NOT_FOUND }
+  )
+  const cowSdk1 = new CowSdk(chainId)
+  cowSdk1.updateChainId(1)
+
+  try {
+    await cowSdk1.cowApi.getProfileData('unexistingAddress')
+  } catch (e) {
+    const error = e as CowError
+    expect(error.message).toEqual("You've passed an invalid URL")
+    expect(fetchMock).toHaveBeenCalledTimes(1)
+    expect(fetchMock).toHaveBeenCalledWith('https://api.cow.fi/affiliate/api/v1/profile/unexistingAddress', {
+      body: undefined,
+      headers: {
+        'Content-Type': 'application/json',
+        'X-AppId': '0x0000000000000000000000000000000000000000000000000000000000000000',
+      },
+      method: 'GET',
+    })
+  }
+})
+
+test('Invalid: Get Profile Data from not supported network', async () => {
+  const profileData = await cowSdk.cowApi.getProfileData('0x6810e776880c02933d47db1b9fc05908e5386b96')
+  expect(profileData).toBeNull()
+})
+
+test('Valid: Sign Order', async () => {
+  gpV2Contracts.signOrder.mockImplementation(() => ({
+    scheme: SigningScheme.EIP712,
+    data: '0x4d306ce7c770d22005bcfc00223f8d9aaa04e8a20099cc986cb9ccf60c7e876b777ceafb1e03f359ebc6d3dc84245d111a3df584212b5679cb5f9e6717b69b031b',
+  }))
+  const order: Omit<UnsignedOrder, 'appData'> = {
+    kind: OrderKind.SELL,
+    partiallyFillable: false, // Allow partial executions of an order (true would be for a "Fill or Kill" order, which is not yet supported but will be added soon)
+    sellToken: '0xc778417e063141139fce010982780140aa0cd5ab', // WETH
+    buyToken: '0x4dbcdf9b62e891a7cec5a2568c3f4faf9e8abe2b', // USDC
+    sellAmount: '1234567890',
+    buyAmount: '1234567890',
+    validTo: 2524608000,
+    receiver: '0x6810e776880c02933d47db1b9fc05908e5386b96',
+    feeAmount: '1234567890',
+  }
+
+  const signedOrder = await cowSdk.signOrder(order)
+  expect(signedOrder.signature).toEqual(SIGNED_ORDER_RESPONSE.signature)
+  expect(signedOrder.signingScheme).toEqual(SIGNED_ORDER_RESPONSE.signingScheme)
+})
+
+test('Valid: Sign cancellation Order', async () => {
+  gpV2Contracts.signOrderCancellation.mockImplementation(() => ({
+    scheme: SigningScheme.EIP712,
+    data: '0x4d306ce7c770d22005bcfc00223f8d9aaa04e8a20099cc986cb9ccf60c7e876b777ceafb1e03f359ebc6d3dc84245d111a3df584212b5679cb5f9e6717b69b031b',
+  }))
+
+  const signCancellationOrder = await cowSdk.signOrderCancellation('validOrderId')
+  expect(signCancellationOrder.signature).toEqual(SIGNED_ORDER_RESPONSE.signature)
+  expect(signCancellationOrder.signingScheme).toEqual(SIGNED_ORDER_RESPONSE.signingScheme)
```

Comment:
If we actually sign an order/orderCancellation, I'd rather just check that signature and signingScheme are present